src/Framework/AspNetCoreAnalyzers/src/Analyzers/Http/HeaderDictionaryAddAnalyzer.cs

```diff
@@ -22,20 +23,15 @@ public override void Initialize(AnalysisContext context)
 
     private static void OnCompilationStart(CompilationStartAnalysisContext context)
     {
-        var symbols = new HeaderDictionarySymbols(context.Compilation);
-
-        if (!symbols.HasRequiredSymbols)
-        {
-            return;
-        }
+        var wellKnownTypes = WellKnownTypes.GetOrCreate(context.Compilation);
 
         context.RegisterOperationAction(context =>
         {
             var invocation = (IInvocationOperation)context.Operation;
 
-            if (SymbolEqualityComparer.Default.Equals(symbols.IHeaderDictionary, invocation.Instance?.Type)
-                && IsAddMethod(invocation.TargetMethod)
-                && invocation.TargetMethod.Parameters.Length == 2)
+            if (IsAddMethod(invocation.TargetMethod)
```

Comment:
Moved `IsAddMethod` first in the if statement. It's cheaper to filter out methods using that method than using SymbolEqualityComparer.